### PR TITLE
[googleCalendar@javahelps.com] Updated xdate.js: avoid error messages in ~/.xsession-errors

### PR DESCRIPTION
**Pull request by claudiux**
Hi @slgobinath,

I propose two minor changes in `lib/xdate.js` to avoid noisy messages in `~/.xsession-errors`.

Tested for a few days, your fabulous desklet still works very well with these changes.

Kind regards.
Claudiux

### CODE REVIEW

File: googleCalendar@javahelps.com/files/googleCalendar@javahelps.com/lib/xdate.js [hunk 1]
```diff
@@ -98,7 +98,7 @@ var XDate = function(t, e, n, r) {
 
     function T(e, n, r) {
         for (var i, o = u.parsers, s = 0; s < o.length; s++)
-            if (i = o[s](e, n, r)) return i;
+            if (i == o[s](e, n, r)) return i;
```
Comment (slgobinath) [inline googleCalendar@javahelps.com/files/googleCalendar@javahelps.com/lib/xdate.js:101]: Hi @claudiux 
With this change, my calendar events are shown one day earlier. My program on Monday is being displayed as on Sunday. Once I changed this line back to `if (i = o[s](e, n, r)) return i;` it works fine.

Comment (slgobinath) [reply googleCalendar@javahelps.com/files/googleCalendar@javahelps.com/lib/xdate.js:101]: Just noticed that there is a related issue https://github.com/linuxmint/cinnamon-spices-desklets/issues/459

Comment (claudiux) [reply googleCalendar@javahelps.com/files/googleCalendar@javahelps.com/lib/xdate.js:101]: Hi @slgobinath,

Please try replacing the function T by this one:
```
    function T(e, n, r) {
        for (var i, o = u.parsers, s = 0; s < o.length; s++) {
            i = o[s](e, n, r);
            if (i) return i;
        }
        r[0] = new t(e);
        return r[0], r
    }
```

Comment (slgobinath) [reply googleCalendar@javahelps.com/files/googleCalendar@javahelps.com/lib/xdate.js:101]: Hi @claudiux 
That works. I already sent a PR https://github.com/linuxmint/cinnamon-spices-desklets/pull/460

**Pull request merged**